include/picongpu/param/memory.param

```diff
@@ -82,6 +82,21 @@ namespace picongpu
         static constexpr uint32_t BYTES_EXCHANGE_Z = 1 * 1024 * 1024; // 1 MiB
         static constexpr uint32_t BYTES_EDGES = 32 * 1024; // 32 kiB
         static constexpr uint32_t BYTES_CORNER = 8 * 1024; // 8 kiB
+
+        /** Reference local domain size
+         *
+         * The size of the local domain for which the exchange sizes `BYTES_*` are configured for.
+         * The required size of each exchange will be calculated at runtime based on the local domain size and the
+         * reference size. The exchange size will be scaled only up and not down. Zero means that there is no reference
+         * domain size, exchanges will not be scaled.
+         */
+        using REF_LOCAL_DOM_SIZE = mCT::Int<0, 0, 0>;
+        /** Scaling rate per direction.
+         *
+         * 1.0 means it scales linear with the ratio between the local domain size at runtime and the reference local
+         * domain size.
+         */
+        static constexpr std::array<float_X, 3> DIR_SCALING_FACTOR = {0.0, 0.0, 0.0};
```

Comment:
looks like we hitting an error with gcc5: https://gitlab.com/hzdr/crp/picongpu/-/jobs/897418670#L770
```
main.cpp:(.text._ZNK8picongpu9ParticlesIN5pmacc4meta6StringIJLc112ELc104ELc0ELc0ELc0ELc0ELc0ELc0ELc0ELc0ELc0ELc0ELc0ELc0ELc0ELc0ELc0ELc0ELc0ELc0ELc0ELc0ELc0ELc0ELc0ELc0ELc0ELc0ELc0ELc0ELc0ELc0ELc0ELc0ELc0ELc0ELc0ELc0ELc0ELc0ELc0ELc0ELc0ELc0ELc0ELc0ELc0ELc0ELc0ELc0ELc0ELc0ELc0ELc0ELc0ELc0ELc0ELc0ELc0ELc0ELc0ELc0ELc0ELc0ELc0EEEEN5boost3mpl6v_itemINS_11chargeRatioINS_18ChargeRatioPhotonsENS1_13pmacc_isAliasEEENS7_INS_9massRatioINS_16MassRatioPhotonsESA_EENS7_INS_13interpolationINS_28FieldToParticleInterpolationINS_9particles6shapes3TSCENS_30AssignedTrilinearInterpolationEEESA_EENS7_INS_5shapeISJ_SA_EENS7_INS_14particlePusherINSH_6pusher6PhotonESA_EENS6_7vector0IN4mpl_2naEEELi0EEELi0EEELi0EEELi0EEELi0EEENS7_INS_13momentumPrev1ENS7_INS_10particleIdENS7_INS_9weightingENS7_INS_8momentumENS7_INS_8positionINS_12position_picESA_EESW_Li0EEELi0EEELi0EEELi0EEELi0EEEE18exchangeMemorySizeEj[_ZNK8picongpu9ParticlesIN5pmacc4meta6StringIJLc112ELc104ELc0ELc0ELc0ELc0ELc0ELc0ELc0ELc0ELc0ELc0ELc0ELc0ELc0ELc0ELc0ELc0ELc0ELc0ELc0ELc0ELc0ELc0ELc0ELc0ELc0ELc0ELc0ELc0ELc0ELc0ELc0ELc0ELc0ELc0ELc0ELc0ELc0ELc0ELc0ELc0ELc0ELc0ELc0ELc0ELc0ELc0ELc0ELc0ELc0ELc0ELc0ELc0ELc0ELc0ELc0ELc0ELc0ELc0ELc0ELc0ELc0ELc0ELc0EEEEN5boost3mpl6v_itemINS_11chargeRatioINS_18ChargeRatioPhotonsENS1_13pmacc_isAliasEEENS7_INS_9massRatioINS_16MassRatioPhotonsESA_EENS7_INS_13interpolationINS_28FieldToParticleInterpolationINS_9particles6shapes3TSCENS_30AssignedTrilinearInterpolationEEESA_EENS7_INS_5shapeISJ_SA_EENS7_INS_14particlePusherINSH_6pusher6PhotonESA_EENS6_7vector0IN4mpl_2naEEELi0EEELi0EEELi0EEELi0EEELi0EEENS7_INS_13momentumPrev1ENS7_INS_10particleIdENS7_INS_9weightingENS7_INS_8momentumENS7_INS_8positionINS_12position_picESA_EESW_Li0EEELi0EEELi0EEELi0EEELi0EEEE18exchangeMemorySizeEj]+0x4f): 
undefined reference to `picongpu::DefaultExchangeMemCfg::DIR_SCALING_FACTOR'
```

I found this on stackoverflow: https://stackoverflow.com/a/28846608

I will change this to a non `constexpr` member and instantiate the struct before getting `DIR_SCALING_FACTOR`. Since it is not performance critical there should be no problem to do it.

Reply:
I think I understand this odr usage issue: we faced it before, I also faced it with the alpaka OpenMP schedule PR. Nothing against your workaround